doc/design/publish-1.md

```diff
@@ -0,0 +1,141 @@
+# Moving Parts
+
+This is a summary of system components that are relevant to publish and how they relate to eachother.
+
+### Dandiset
+A DANDI dataset.
+Each dandiset is guaranteed to have a draft Version, and can also have any number of published Versions.
+
+### Version
+Versions consist of a JSON blob of metadata and any number of Assets.
+There are two kinds of Version: draft and published.
+Published Versions are immutable, only draft Versions are editable (except admins, who have superpowers).
+Versions also track their validation status.
+
+### AssetBlob
+This is basically just a pointer to a blob in S3.
+AssetBlobs are central to how we implement deduplication of blobs in S3.
+Multiple Assets can refer to the same AssetBlob, meaning we only need to store each blob in S3 once.
```

Comment:
```suggestion
Multiple Assets can refer to the same AssetBlob, meaning we only need to store each blob in S3 once.
Each Asset could have a different metadata record for the same AssetBlob.
```
if we are to clarify, I think this might be valuable to add